internal/api/core/submission.go

```diff
@@ -0,0 +1,64 @@
+package core
+
+import (
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type BaseSubmitResponse struct {
+	Rejected bool   `json:"rejected"`
+	Message  string `json:"message"`
+
+	GradingSuccess bool               `json:"grading-success"`
+	GradingInfo    *model.GradingInfo `json:"result"`
+}
+
+func GradeToSubmissionResponse(request APIRequestAssignmentContext, submissionPath string, email string, message string, options grader.GradeOptions) BaseSubmitResponse {
```

Comment:
The name feel a bit off.
I feel like names like this usually imply a conversion, but we are doing a full grading (which is conceptually heavier).
Maybe something like "GradeRequestSubmission"?